tests/Connection/ExceptionHandlingTest.php

```diff
@@ -33,29 +34,37 @@ protected function setUp(): void
 
     public function testDriverExceptionDuringQueryAcceptsBinaryData(): void
     {
+        $driverException = $this->createMock(DriverException::class);
+
         $this->exceptionConverter->expects(self::once())
             ->method('convert')
-            ->with(self::stringContains('with params ["ABC", "\x80"]'));
+            ->willReturn(new DBALDriverException('', $driverException));
 
-        $this->connection->convertExceptionDuringQuery(
-            $this->createMock(DriverException::class),
+        $convertedException = $this->connection->convertExceptionDuringQuery(
+            $driverException,
             '',
             ['ABC', chr(128)]
         );
+
+        self::assertSame(['ABC', chr(128)], $convertedException->getQueryParams());
     }
 
     public function testDriverExceptionDuringQueryAcceptsResource(): void
     {
+        $driverException = $this->createMock(DriverException::class);
+
         $this->exceptionConverter->expects(self::once())
             ->method('convert')
-            ->with(self::stringContains('Resource'));
+            ->willReturn(new DBALDriverException('', $driverException));
 
-        $this->connection->convertExceptionDuringQuery(
-            $this->createMock(DriverException::class),
+        $convertedException = $this->connection->convertExceptionDuringQuery(
+            $driverException,
             'INSERT INTO file (`content`) VALUES (?)',
             [
                 1 => fopen(__FILE__, 'r'),
             ]
         );
+
+        self::assertIsResource($convertedException->getQueryParams()[1]);
```

Comment:
I've refactored these tests, but I'm not sure if they really bring any value now. We should probably just create a test that ensures that the params & types are properly passed to the `DriverException` and that's it.

Reply:
I've removed this test altogether. The `getQuery()` part is tested elsewhere.